tests/proc/lif/test_models.py

```diff
@@ -0,0 +1,350 @@
+        current decay of 0. Voltage state variable increases from 128. to 255.
+        with steps of 64., 32., 16., etc.
+        """
+        shape = (1,)  # a single neuron
+        num_steps = 8
+        # send activation of 128. at timestep = 1
+        sps = VecSendProcess(shape=shape, num_steps=num_steps,
+                             vec_to_send=(2**7)*np.ones(shape, dtype=np.float),
+                             send_at_times=np.array([True, False, False,
+                                                     False, False, False,
+                                                     False, False]))
+        lif = LIF(shape=shape,
+                  du=0,
+                  dv=0.5,  # decay_v = 1/2
+                  bias=np.zeros(shape, dtype=np.float),
+                  bias_exp=np.ones(shape, dtype=np.float),
+                  vth=256.)
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.out_ports.s_out.connect(lif.in_ports.a_in)
+        lif.out_ports.s_out.connect(spr.in_ports.s_in)
+        rcnd = RunSteps(num_steps=1)
+        rcfg = LifRunConfig(select_tag='floating_pt')
+        lif_v = []
+        # Run 1 timestep at a time and collect state variable u
+        for j in range(num_steps):
+            lif.run(condition=rcnd, run_cfg=rcfg)
+            lif_v.append(lif.v.get()[0])
+        lif.stop()
+        expected_v_timeseries = [128., 192., 224., 240., 248., 252., 254., 255.]
+        print(expected_v_timeseries)
+        self.assertListEqual(expected_v_timeseries, lif_v)
+
+    def test_bitacc_pm_no_decay(self):
+        """
+        Set up bias = 2 * 2**6 = 128 and threshold = 8<<6 Without any current
+        and voltage decay and in the absence of external input, bias driven
+        neurons spike every 4th time-step.
+        """
+        shape = (10,)
+        num_steps = 10
+        sps = VecSendProcess(shape=shape, num_steps=num_steps,
+                             vec_to_send=np.zeros(shape, dtype=np.int16),
+                             send_at_times=np.ones((num_steps,), dtype=np.bool))
+        lif = LIF(shape=shape,
+                  du=0, dv=0,
+                  bias=2*np.ones(shape, dtype=np.int32),
+                  bias_exp=6*np.ones(shape, dtype=np.int32),
+                  vth=8)
+        spr = VecRecvProcess(shape=(num_steps, shape[0]))
+        sps.out_ports.s_out.connect(lif.in_ports.a_in)
+        lif.out_ports.s_out.connect(spr.in_ports.s_in)
+        rcnd = RunSteps(num_steps=num_steps)
+        rcfg = LifRunConfig(select_tag='fixed_pt')
+        lif.run(condition=rcnd, run_cfg=rcfg)
+        spk_data_through_run = spr.spk_data.get()
+        lif.stop()
+        expected_spk_data = np.zeros((num_steps, shape[0]))
+        expected_spk_data[3:10:4, :] = 1
+        self.assertTrue(np.all(expected_spk_data == spk_data_through_run))
+
+    def test_bitacc_pm_impulse_du(self):
+        """
+        Send a single activation of magnitude 128 with a current decay of
```

Comment:
Start the docstring with "Test of Loihi 1 bit accurate LIF dynamics with external input and a decay of the integrated current. "

Reply:
Done.